service/history/transferQueueProcessor.go

```diff
@@ -630,6 +630,28 @@ func (t *transferQueueProcessorImpl) processSignalExecution(task *persistence.Tr
 		return nil
 	}
 
+	// handle workflow signal itself
```

Comment:
Probably you want to do the check for IsWorkflowExecutionRunning before even hitting any processing for signal.  Also make sure to perform this check in childworkflow transfer task processing.